jaxrs/src/main/java/com/facebook/airlift/jaxrs/JerseyAuthorizationFilter.java

```diff
@@ -0,0 +1,110 @@
+
+import com.facebook.airlift.http.server.AuthorizationResult;
+import com.facebook.airlift.http.server.Authorizer;
+import com.facebook.airlift.http.server.HttpServerConfig;
+import com.google.common.collect.ImmutableSet;
+import com.google.inject.Inject;
+
+import javax.annotation.security.RolesAllowed;
+import javax.ws.rs.container.ContainerRequestContext;
+import javax.ws.rs.container.ContainerRequestFilter;
+import javax.ws.rs.container.ResourceInfo;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.ext.Provider;
+
+import java.security.Principal;
+import java.util.Optional;
+import java.util.Set;
+
+import static java.lang.String.format;
+
+@Provider
+public class JerseyAuthorizationFilter
+        implements ContainerRequestFilter
+{
+    private final Authorizer authorizer;
+    private final HttpServerConfig httpServerConfig;
+
+    @Context
+    private ResourceInfo resourceInfo;
+
+    @Inject
+    public JerseyAuthorizationFilter(Authorizer authorizer, HttpServerConfig httpServerConfig)
+    {
+        this.authorizer = authorizer;
+        this.httpServerConfig = httpServerConfig;
+    }
+
+    @Override
+    public void filter(ContainerRequestContext request)
+    {
+        Principal principal = request.getSecurityContext().getUserPrincipal();
+        if (principal == null) {
+            request.abortWith(Response.status(Response.Status.FORBIDDEN)
+                    .entity("Request principal is missing.")
+                    .build());
+            return;
+        }
+
+        Optional<Set<String>> allowedRoles = getAllowedRoles();
+        if (isEndpointUnmarked(allowedRoles)) {
+            switch (httpServerConfig.getDefaultAuthorizationPolicy()) {
+                case ALLOW:
+                    return;
+                case DENY:
+                    request.abortWith(Response.status(Response.Status.FORBIDDEN)
+                            .entity(format("Principal %s is not allowed to access the resource. Reason: denied by default policy",
+                                    principal.getName()))
+                            .build());
+                    return;
+                case DEFAULT_ROLES:
+                    allowedRoles = Optional.of(httpServerConfig.getDefaultAllowedRoles());
+                    break;
+            }
+        }
+
+        AuthorizationResult result = authorizer.authorize(principal, allowedRoles.get());
+        if (result == null || !result.isAllowed()) {
+            request.abortWith(Response.status(Response.Status.FORBIDDEN)
+                    .entity(format("Principal %s is not allowed to access the resource. Reason: %s",
+                            principal.getName(),
+                            result.getReason().orElse("")))
+                    .build());
+        }
+    }
+
+    private Optional<Set<String>> getAllowedRoles()
+    {
+        if (resourceInfo.getResourceMethod().isAnnotationPresent(RolesAllowed.class)) {
```

Comment:
Adding some comments here might be helpful.
something like
//Checking Method level annotation for roles.
//Checking Class level annotation for roles

Reply:
In my personal opinion - the code is very clear and we don't need comments here.